### ISSUE TITLE
linebreak-style doesn’t detect multiple problems per file

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#3981]
The `linebreak-style` rule only reports the first problem that was detected, even if there are more.

When running following command

```
$ echo "var foo;\nvar bar;\nvar baz;" | ./node_modules/.bin/eslint --no-eslintrc --rule 'linebreak-style: [2, "windows"]' --stdin
```

I get the following output:

```
<text>
  1:9  error  Expected linebreaks to be 'CRLF' but found 'LF'  linebreak-style

✖ 1 problem (1 error, 0 warnings)
```

I would expect 2 problems.

ESLint version: `1.5.1`

**Comment by michaelficarra** [comment]
So if the wrong line terminators are used throughout the file (almost certainly the most common case), you want every single line to be reported?

**Comment by nzakas** [comment]
IIRC we did this intentionally because otherwise you would get spammed if the entire file is formatted incorrectly. I'm not sure it's a good idea to do that.

**Comment by lo1tuma** [comment]
I see. If we want to implement auto fixing for this rule I think we need to
report every problem.
So we need to decide what would be more valuable.

**Comment by alberto** [comment]
I'm sorry, I didn't realize this was not already accepted.

**Comment by nzakas** [comment]
Good point. We should try to get some input from our users before going forward. 

To summarize: right now, we only warn on the first instance of an incorrect line break, assuming that if there's one problem, you'll reformat the whole file. However, auto-fixing is tied to individual lint messages, so we must have a message for each fix in case the fix cannot be applied.

To make this rule auto fix, we would need to warn on every instance of an incorrect newline, which could mean a lot of messages in a file that has is formatted incorrectly.

So the question is: do we start warning for every incorrect newline so we can do auto fix? Or do we leave as is?

**Comment by alberto** [comment]
It could also be a configuration option for the rule.

**Comment by nzakas** [comment]
That would be hard to manage, as those who want auto-fixing might not realize the difference between the two modes.

**Comment by alberto** [comment]
Well, that's what the docs are for. ;)

Just for reference, I checked JSCS and they have a `reportOncePerFile` config option: http://jscs.info/rule/validateLineBreaks.html

**Comment by alberto** [comment]
Another option would be to report all the errors only when in auto-fix mode (if there is a way to check that).

**Comment by hzoo** [comment]
Maybe postpone warning on every instance until autofix is implemented?

**Comment by nzakas** [comment]
@hzoo that doesn't really help because it's easy to implement auto-fixing.

**Comment by ilyavolodin** [comment]
I like @alberto suggestion about reporting all errors only when `--fix` flag is passed in. I don't think I would want to see an error per line in the error report.

**Comment by nzakas** [comment]
That would mean a rule behaves differently based on the mode - are we comfortable with that?

**Comment by ilyavolodin** [comment]
Rules already behave somewhat differently based on `--fix` flag (at least from users' perspective) I think it should be fine.

**Comment by hzoo** [comment]
Oh ok I don't why I was assuming everyone would use `--fix` for some reason. I would agree with @alberto's suggestion as well then.

**Comment by nzakas** [comment]
@ilyavolodin they don't behave differently now, they always report the same messages (minus fixed messages). In this case, the rule could potentially return many more messages than without `--fix`. That's my concern, especially if, for some reason, all line breaks cannot be fixed.

**Comment by alberto** [comment]
@nzakas In my opinion the rule is behaving differently now to the rest of the rules (I don't know if there is any other rule that reports just once per file). So it is already a special case. The --fix would behave as any other rule. Is there any case where you anticipate not all breaks could be fixed?

**Comment by nzakas** [comment]
Not really the point, the point is having the rule behave differently with --fix vs. not. That's confusing. 

I'm thinking now that we have --fix, it's less annoying to have the rule warn on every instance since you can fix it easily. Thoughts?

**Comment by alberto** [comment]
I would be problematic for people running it from an IDE, since fix is not exposed to them, is it?

We could add it as an option to report (and fix) all the problems.

**Comment by nzakas** [comment]
Most IDEs have an easy way to switch line endings, so I'm not too worried about that situation.

**Comment by alberto** [comment]
Then I say yes

**Comment by nzakas** [comment]
Cool, let's do it.

**Comment by alberto** [comment]
Hmm, this is text based, so I don't know how to get the range of the linebreak from the loc, to create each fix. Any ideas?

**Comment by nzakas** [comment]
See https://github.com/eslint/eslint/blob/master/lib/rules/no-multi-spaces.js as an example
